microprofile/bean-validation/src/main/java/io/helidon/microprofile/bean/validation/HelidonConstraintViolationExceptionMapper.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2022 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.microprofile.bean.validation;
+
+import javax.validation.ConstraintViolationException;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.ExceptionMapper;
+import javax.ws.rs.ext.Provider;
+
+
+/**
+ * A JAX-RS provider that maps {@link javax.validation.ConstraintViolationException} from bean validation
+ * to a proper JAX-RS response with {@link javax.ws.rs.core.Response.Status#BAD_REQUEST} status.
+ * If this provider is not present, validation exception from Validation would end with an internal server
+ * error.
+ */
+@Provider
+public class HelidonConstraintViolationExceptionMapper implements ExceptionMapper<ConstraintViolationException> {
+
+    /**
+     * Return Validation Exception, wrapped as a bad request.
+     * @param exception Validation exception
+     * @return BAR_REQUEST Response.
+     */
+    @Override
+    public Response toResponse(ConstraintViolationException exception) {
+        return Response.status(Response.Status.BAD_REQUEST)
```

Comment:
I saw the comment from Tomas ... the JAX-RS spec has some rules on how to map these. Generally, 400 makes sense if the error is found while validating the input (request) but not so much after that. If the application has some other logic that results in an error (after request was _consumed_), then 500 would be more appropriate. I would probably let the developer decide and not include this mapper as part of Helidon.

Reply:
I have actually requested this as part of this PR, as throwing a 500 is not the best reply (esp. when this is a client error, not a server error - the data is in invalid format).

Reply:
We have decided to 400 for JAX-RS responses.